modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Controller/PersonifyAuthController.php

```diff
@@ -0,0 +1,282 @@
+<?php
+
+namespace Drupal\openy_gc_auth_personify\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Logger\LoggerChannelFactory;
+use Drupal\Core\Routing\TrustedRedirectResponse;
+use Drupal\Core\Url;
+use Drupal\personify\PersonifyClient;
+use Drupal\personify\PersonifySSO;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Personify controller to handle Personify SSO authentication.
+ */
+class PersonifyAuthController extends ControllerBase {
+
+  /**
+   * Personify SSO service.
+   *
+   * @var \Drupal\personify\PersonifySSO
+   */
+  protected $personifySSO;
+
+  /**
+   * Personify Client service.
+   *
+   * @var \Drupal\personify\PersonifyClient
+   */
+  protected $personifyClient;
+
+  /**
+   * Request stack.
+   *
+   * @var \Symfony\Component\HttpFoundation\Request|null
+   */
+  protected $request;
+
+  /**
+   * Logger interface.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $logger;
+
+  /**
+   * Config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * PersonifyAuthController constructor.
+   *
+   * @param \Drupal\personify\PersonifySSO $personifySSO
+   *   Personify SSO service.
+   * @param \Drupal\personify\PersonifyClient $personifyClient
+   *   Personify Client service.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   Config factory.
+   * @param \Symfony\Component\HttpFoundation\RequestStack $requestStack
+   *   Request stack.
+   * @param \Drupal\Core\Logger\LoggerChannelFactory $loggerChannelFactory
+   *   Logger factory.
+   */
+  public function __construct(
+    PersonifySSO $personifySSO,
+    PersonifyClient $personifyClient,
+    ConfigFactoryInterface $configFactory,
+    RequestStack $requestStack,
```

Comment:
There is no need to add this service. Every controller method receive's Request object as first parameter by default.